tools/Generator.CLI/SourceGenerator/TaskTupleAwaiterGenerator.cs

```diff
@@ -0,0 +1,196 @@
+using System.Text;
+using static Generator.CLI.SourceGenerator.CommonTokens;
+
+namespace Generator.CLI.SourceGenerator
+{
+	public static class TaskTupleAwaiterGenerator
+	{
+		public static string GenerateContent(int arity)
+		{
+			/**
+			 * based on https://github.com/buvinghausen/TaskTupleAwaiter/blob/master/src/TaskTupleAwaiter/TaskTupleExtensions.cs
+			 * https://github.com/buvinghausen/TaskTupleAwaiter/commit/ead4851edaf7553fc938a22377e48ff9f5c70cdc
+			 */
+			var sb = new StringBuilder();
+			AddFileHeader(sb);
+			AddArityOne(sb);
+			if (arity > 1)
+			{
+				for (var i = 2; i < arity; i++)
+				{
+					AddGeneratedArity(sb, i);
+				}
+
+				sb.AppendLine($"#region Task");
+				for (var i = 2; i < arity; i++)
+				{
+					AddTasks(sb, i);
+				}
+				sb.AppendLine($"#endregion Task");
+			}
+
+			// namespace close brace
+			sb.Append("}");
```

Comment:
This one is the class's closing brace, right?

Reply:
Well you're truly thorough. Even comments are inspected for correctness - fixed :)

Reply:
Sorry -_- hope that's not problem.